dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/ApplicationProcessor.java

```diff
@@ -254,6 +265,11 @@ private OpenAPI processWebModule(Container appContainer, WebModuleInfo moduleInf
             OASFilter oasFilter = OpenAPIUtils.getOASFilter(appClassloader, configProcessor.getOpenAPIFilterClassName());
             if (oasFilter != null) {
                 final OpenAPIFilter filter = new OpenAPIFilter(oasFilter);
+
+                ThreadContextAccessor tca = System.getSecurityManager() == null ? ThreadContextAccessor.getThreadContextAccessor() : AccessController.doPrivileged(getThreadContextAccessorAction);
```

Comment:
Does this need to be done every time or could it be stored in a static?

Reply:
Can we follow the examples in `ThreadContextAccessor`?

```
private static final ThreadContextAccessor threadContextAccessor = AccessController.doPrivileged(ThreadContextAccessor.getPrivilegedAction());
```

(but uppercased ;) )

and then

```
 Object origCL = threadContextAccessor.pushContextClassLoaderForUnprivileged(cl);
 try {
     ...
 } finally {
     threadContextAccessor.popContextClassLoaderForUnprivileged(origCL);
 }
```

Reply:
Done ;-)